lib/galaxy/tool_util/lint.py

```diff
@@ -89,59 +124,67 @@ def __init__(self, level, skip_types=None, object_name=None):
         self.skip_types = skip_types or []
         self.level = level
         self.object_name = object_name
-        self.found_errors = False
-        self.found_warns = False
+        self.message_list = []
```

Comment:
```suggestion
        self.message_list: List[LintMessage] = []
```
For this, we will need to import `from typing import List` but again it is nicer to be explicit about the type here

Reply:
I added type annotations at most places. Maybe you can recheck?

Reply:
Thank you! That's a lot of type hints :laughing: 
Check `mypy` errors, usually, they give a nice hint to fix them.
I would use just `None` instead of `NoReturn` for methods. 
As per the documentation:
> `NoReturn` is a special type to annotate functions that never return normally. For example, a function that unconditionally raises an exception.